dpnp/dpnp_iface_statistics.py

```diff
@@ -453,47 +459,178 @@ def corrcoef(x, y=None, rowvar=True, *, dtype=None):
     return out
 
 
-def correlate(x1, x2, mode="valid"):
-    """
+def _get_padding(a_size, v_size, mode):
+    if v_size > a_size:
+        a_size, v_size = v_size, a_size
+
+    if mode == "valid":
+        l_pad, r_pad = 0, 0
+    elif mode == "same":
+        l_pad = v_size // 2
+        r_pad = v_size - l_pad - 1
+    elif mode == "full":
+        l_pad, r_pad = v_size - 1, v_size - 1
+    else:
+        raise ValueError(
+            f"Unknown mode: {mode}. Only 'valid', 'same', 'full' are supported."
+        )
+
+    return l_pad, r_pad
+
+
+def _run_native_sliding_dot_product1d(a, v, l_pad, r_pad):
+    queue = a.sycl_queue
+
+    usm_type = dpu.get_coerced_usm_type([a.usm_type, v.usm_type])
+    out_size = l_pad + r_pad + a.size - v.size + 1
+    out = dpnp.empty(
+        shape=out_size, sycl_queue=queue, dtype=a.dtype, usm_type=usm_type
```

Comment:
I think we need to leave somewhere comment that result type is always expected as a type of input arrays.

Reply:
added